iceberg/iceberg-catalog/pom.xml

```diff
@@ -69,7 +69,6 @@
     <dependency>
       <groupId>org.apache.hive</groupId>
       <artifactId>hive-exec</artifactId>
-      <scope>test</scope>
```

Comment:
please restore the test scope. Iceberg catalog shouldn't have a cyclic dependency on `hive-exec`

Reply:
Zoltan kept this dependency because `HMSTablePropertyHelper` depends on this two:
```
import org.apache.hadoop.hive.ql.parse.TransformSpec;
import org.apache.hadoop.hive.ql.session.SessionStateUtil;
```

Also, `hive-exec` doesn't depend on any iceberg modules and other iceberg modules (iceberg-handler) depends on `hive-exec`, are you sure we can't have it?

Reply:
In the existing examples in hive code, with these inputs, we convert hive partition fields schema to iceberg identity partition spec, like in `HiveSchemaUtil`:

```
  public static PartitionSpec spec(Schema schema, List<FieldSchema> fieldSchemas) {
    PartitionSpec.Builder builder = PartitionSpec.builderFor(schema);
    fieldSchemas.forEach(fieldSchema -> builder.identity(fieldSchema.getName().toLowerCase()));
    return builder.build();
  }
```
But as I understand we don't want to convert to identity spec, but the FieldSchema doesn't contain all the details about transforms. How can we do it?

Reply:
@difin try this
````
List<FieldSchema> cols = Lists.newArrayList(table.getSd().getCols());
Properties catalogProperties = HMSTablePropertyHelper.getCatalogProperties(table);
Schema schema = HiveSchemaUtil.convert(cols, true);

PartitionSpec spec = PartitionSpecParser.fromJson(
  schema, table.getParameters().get(InputFormatConfig.PARTITION_SPEC))
````

Reply:
Hi @deniskuzZ ,
Currently `HiveIcebergMetaHook` calculates `PARTITION_SPEC` and then sets it as a table property, if we do this change then instead of calculating the value it will try to get it from table properties and will be missing.

Reply:
Hi @deniskuzZ ,
Currently `HiveIcebergMetaHook#preCreateTable` calculates `PARTITION_SPEC` and puts it into HMS table properties, if we do this change then instead of calculating the value it will try to get it from table properties and it will be missing because it hadn't been calculated yet.

Reply:
Thanks, done.
Removed runtime dependency on hive-exec from iceberg catalog.